dragonfly/block/trapdoor.go

```diff
@@ -0,0 +1,135 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/wood"
+	"github.com/df-mc/dragonfly/dragonfly/entity/physics"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math"
+)
+
+// Trapdoor is a block that can be used as an openable 1x1 barrier
+type Trapdoor struct {
```

Comment:
Trapdoors don't diffuse light so you should implement the LightDiffuser interface and `return 0` for the diffusion level.